tests/VersionParserTest.php

```diff
@@ -86,8 +86,12 @@ public function successfulNormalizedVersions()
             'parses arbitrary/3' => array('dev-feature/foo', 'dev-feature/foo'),
             'ignores aliases' => array('dev-master as 1.0.0', '9999999-dev'),
             'semver metadata' => array('dev-master+foo.bar', '9999999-dev'),
-            'semver metadata/2' => array('1.0.0-beta.5+foo', '1.0.0.0-beta5'),
+            'semver metadata/2' => array('1.0.0-beta.5+foo', '1.0.0.0-beta.5'),
             'semver metadata/3' => array('1.0.0+foo', '1.0.0.0'),
+            'semver metadata/4' => array('1.0.0-alpha.3.1+foo', '1.0.0.0-alpha.3.1'),
```

Comment:
IMO this should normlaize to 1.0.0.0-alpha3.1 to keep the old behavior, otherwise we have the same issue as with the date versions on packagist.

It'll make the patch a bit more complicated though, but if you find a sane way to do it it'd be great I think, then yes I am not against the patch itself although it's not really something I wanna encourage..